packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Thenable} from 'react-reconciler/src/ReactFiberScheduler';
+
+import warningWithoutStack from 'shared/warningWithoutStack';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// Keep in sync with ReactDOMUnstableNativeDependencies.js
+// ReactDOM.js, and ReactTestUtils.js:
+const [
+  /* eslint-disable no-unused-vars */
+  getInstanceFromNode,
+  getNodeFromInstance,
+  getFiberCurrentPropsFromNode,
+  injectEventPluginsByName,
+  eventNameDispatchConfigs,
+  accumulateTwoPhaseDispatches,
+  accumulateDirectDispatches,
+  enqueueStateRestore,
+  restoreStateIfNeeded,
+  dispatchEvent,
+  runEventsInBatch,
+  /* eslint-enable no-unused-vars */
+  flushPassiveEffects,
+] = ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Events;
+
+const RendererHelpers = {
+  flushPassiveEffects,
+  batchedUpdates: ReactDOM.unstable_batchedUpdates,
+};
+
+const {
+  isActingUpdates,
+} = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
+
+// the rest of this file should be exactly the same in ReactTestUtilsAct.js,
+// ReactTestRendererAct.js, ReactNoopAct.js, ReactNoopPersistentAct.js
+
+let actingUpdatesScopeDepth = 0;
+
+function incrementActingScopeDepth() {
+  actingUpdatesScopeDepth++;
+  isActingUpdates[0] = true;
+}
+
+function decrementActingScopeDepth() {
+  actingUpdatesScopeDepth--;
+  if (actingUpdatesScopeDepth === 0) {
+    isActingUpdates[0] = false;
+  }
+}
+
+let didWarnAboutMessageChannel = false;
+
+let enqueueTask;
```

Comment:
`enqueueTask` is the same across all these implementations, right? Might as well move it to `shared`.

Reply:
The reason not to move it to `shared/` is if you're able to choose a more specific implementation per renderer. Like if we assume that the test renderer always runs in a Node environment.

Reply:
I don't think we can make that guarantee. I'll move it into a `shared/` module.